bip-p2qrh.mediawiki

```diff
@@ -50,26 +50,40 @@ The following table is non-exhaustive, but meant to be informative to the averag
+P2QRH uses a 32-byte HASH256 (specifically SHA-256 x2, which is similar to that used in [https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki#specification BIP-16]) of the public key to reduce the size of new outputs and also to increase security by not having the public key available on-chain. This serves as a minimal cryptographic commitment to a public key. This goes into the output spend script, which does not receive the witness discount. Signatures, however, do receive a 4x witness discount. SQIsign public keys are between 64 and 128 bytes in length, depending on the security level. To promote user adoption and general user-friendliness, the most secure variant of SQIsign (NIST V, 256 bit) is proposed, despite the moderate increase in key length.
 
-In the distant future, following the implementation of P2QR and P2QRH address formats in a soft fork, there will likely be a need for Pay to Quantum Secure (P2QS) addresses. These will require specialized quantum hardware for signing while still being verifiable by classical means. Additional follow-on BIPs will be needed to implement P2QS. However, until specialized quantum cryptography hardware is widespread, quantum resistant addresses should be an adequate intermediate solution.
+Not having public keys exposed on-chain is a major constraint for quantum security, otherwise funds would need to be spent to new addresses on a regular basis in order to prevent the possibility of a "long-range CRQC attack" recovering the key behind high value addresses. A long-range quantum attack can be considered one performed with chain data, such as that from a used address or one encoded in a spend script. A "short-range quantum attack" would be one done on keys in the mempool, which is seen as impractical given transaction throughput and the block time. The more value being sent, the higher the fee should be set, in order to commit the transaction to the chain as soon as possible, thus making useless the public key revealed by spending a UTXO.
+
+Support for FALCON will also be introduced. While its signatures are roughly 4x larger than SQIsign signatures and 20x larger than Schnorr signatures, it will be necessary to have a fallback algorithm should a flaw be found in SQIsign, which is much newer and experimental. FALCON is a more conservative approach to applied lattice cryptography than SQIsign, and its use is approved by NIST. However, even SQIsign signatures are roughly 5x larger than Schnorr signatures. This means, to maintain present transaction throughput, an increase in the witness discount will likely be desired in a QuBit soft fork. That will be specified in a future QuBit BIP.
```

Comment:
I'm curious here.  If we introduce both FALCON and SQIsign, what would be the incentive for a wallet developer to use SQIsign over FALCON?  Wouldn't they want to advertise having 'the best in quantum resistance' and not the 2nd best?

Reply:
SQIsign will be smaller and cheaper.

Reply:
Okay - makes sense I suppose.